test/compress/global_defs.js

```diff
@@ -123,6 +126,8 @@ mixed: {
         console.log(CONFIG);
     }
     expect: {
+        const FOO = { BAR: 0 };
+        console.log(FOO.BAR);
```

Comment:
PTAL at https://github.com/mishoo/UglifyJS2/pull/1469/commits/e88acb47a941c61321a1b3710f15933f269915b2 and see if that matches what you mean?